### ISSUE TITLE
Getting "Unexpected Error" after unpublishing then republishing story

### ISSUE THREAD

**Issue by jwirfs-brock** [denverfoundation/storybase#534]
I'm getting an "Unexpected Error" message for a story that I unpublished then republished through the "My Stories" page:

http://staging.floodlightproject.org/en/stories/dataviz-how-to-make-a-bar-chart-in-fusion-tables/viewer/

This story shows up on the Explore page, and you can view its Story Detail page, but when you try to launch the viewer it returns an "Unexpected Error."

**Comment by ghing** [comment]
This is the stack trace:

```
    Traceback (most recent call last):

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/core/handlers/base.py", line 136, in get_response
        response = response.render()

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/response.py", line 97, in render
        self._set_content(self.rendered_content)

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/response.py", line 75, in rendered_content
        content = template.render(context)

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/base.py", line 123, in render
        return self._render(context)

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/base.py", line 117, in _render
        return self.nodelist.render(context)

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/base.py", line 744, in render
        bits.append(self.render_node(node, context))

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/base.py", line 757, in render_node
        return node.render(context)

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/classytags/core.py", line 102, in render
        return self.render_tag(context, **kwargs)

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/sekizai/templatetags/sekizai_tags.py", line 75, in render_tag
        rendered_contents = nodelist.render(context)

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/base.py", line 744, in render
        bits.append(self.render_node(node, context))

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/base.py", line 757, in render_node
        return node.render(context)

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/defaulttags.py", line 227, in render
        nodelist.append(node.render(context))

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/base.py", line 792, in render
        output = self.filter_expression.resolve(context)

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/base.py", line 510, in resolve
        obj = self.var.resolve(context)

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/base.py", line 653, in resolve
        value = self._resolve_lookup(context)

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/base.py", line 698, in _resolve_lookup
        current = current()

      File "/home/floodlight/webapps/floodlight_staging/floodlight/apps/storybase_story/models.py", line 769, in render
        return getattr(self, "render_" + format).__call__(show_title)

      File "/home/floodlight/webapps/floodlight_staging/floodlight/apps/storybase_story/models.py", line 809, in render_html
        output.append(render_to_string(template_filename, context))

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/loader.py", line 183, in render_to_string
        return t.render(Context(dictionary))

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/base.py", line 123, in render
        return self._render(context)

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/base.py", line 117, in _render
        return self.nodelist.render(context)

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/base.py", line 744, in render
        bits.append(self.render_node(node, context))

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/base.py", line 757, in render_node
        return node.render(context)

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/template/base.py", line 885, in render
        return func(*func_args)

      File "/home/floodlight/webapps/floodlight_staging/floodlight/apps/storybase_story/templatetags/story.py", line 23, in container
        asset = context['assets'].get(container__name=value).asset

      File "/home/floodlight/webapps/floodlight_staging/venv/lib/python2.7/site-packages/django/db/models/query.py", line 351, in get
        % (self.model._meta.object_name, num, kwargs))

    MultipleObjectsReturned: get() returned more than one SectionAsset -- it returned 2! Lookup parameters were {'container__name': u'left'}
```

**Comment by ghing** [comment]
@jwirfs-brock I'm not seeing the error any more.  Did you fix something?  The error had to do with having multiple assets assigned to the same container in the same section, but this no longer seems to be the case. I'd say this is likely caused by editing the same story in 2 separate tabs.  I've created an issue for that as #535.

**Comment by jwirfs-brock** [comment]
I didn't change anything, but I did sign out of my account and closed all
windows (none of which had the Story Builder open).

**Comment by ghing** [comment]
Just a quick note: I'm seeing this error message for both stories that have been published and stories that have never been published, so I'm thinking that the details in Jordan's original report of this happening after a story was unpublished then republished might be a red-herring.

**Comment by ghing** [comment]
A workaround to suppress the "Unexpected error" message was implemented in 60b3028 and deployed into production.  I'm going to close this issue. Any further discussion and addressing the underlying cause should happen in #535.
